### ISSUE TITLE
FTBFS: multiple definition of `status_snapshot' and 'exiting'

### ISSUE THREAD

**Issue by jean-christophe-manciot** [net-snmp/net-snmp#161]
Ubuntu groovy
binutils: 2.35-1ubuntu1 (ships /usr/bin/ld)
gcc: 4:10.1.0-1ubuntu1
libc-bin: 2.31-0ubuntu10
make: 4.3-4ubuntu1
 
snmp tag: v5.8.1.pre2

Building it with:
```
CFLAGS='-g'
./configure     --build=x86_64-pc-linux-gnu \
                --prefix=/usr --sysconfdir=/etc --localstatedir=/var \
                --with-logfile="/var/log/snmpd.log" --with-persistent-directory="/var/net-snmp" \
                --enable-blumenthal-aes \
                --enable-embedded-perl \
                --enable-ipv6 \
                --enable-mib-config-checking \
                --enable-shared \
                --with-default-snmp-version="3" --with-sys-contact="noone@no.where" --with-sys-location="Unknown" \
                --with-libs="-lsensors" \
                --with-mib-modules="agent/extend agent/nsCache agent/nsDebug agent/nsLogging agent/nsModuleTable agent/nsTransactionTable agent/nsVacmAccessTable agent_mibs agentx agentx/agentx_config agentx/client agentx/master agentx/master_admin agentx/protocol agentx/subagent default_modules deliver/deliverByNotify disman/event disman/event/mteEvent disman/event/mteEventConf disman/event/mteEventNotificationTable disman/event/mteEventSetTable disman/event/mteEventTable disman/event/mteObjects disman/event/mteObjectsConf disman/event/mteObjectsTable disman/event/mteScalars disman/event/mteTrigger disman/event/mteTriggerBooleanTable disman/event/mteTriggerConf disman/event/mteTriggerDeltaTable disman/event/mteTriggerExistenceTable disman/event/mteTriggerTable disman/event/mteTriggerThresholdTable disman/expr/expErrorTable disman/expr/expExpression disman/expr/expExpressionConf disman/expr/expExpressionTable disman/expr/expObject disman/expr/expObjectConf disman/expr/expObjectTable disman/expr/expScalars disman/expr/expValue disman/expr/expValueTable disman/expression disman/nslookup-mib disman/nslookup/lookupCtlTable disman/nslookup/lookupResultsTable disman/ping-mib disman/ping/pingCtlTable disman/ping/pingProbeHistoryTable disman/ping/pingResultsTable disman/schedule disman/schedule/schedConf disman/schedule/schedCore disman/schedule/schedTable disman/traceroute-mib disman/traceroute/traceRouteCtlTable disman/traceroute/traceRouteHopsTable disman/traceroute/traceRouteProbeHistoryTable disman/traceroute/traceRouteResultsTable etherlike-mib etherlike-mib/data_access/dot3stats etherlike-mib/data_access/dot3stats_linux etherlike-mib/dot3StatsTable etherlike-mib/dot3StatsTable/dot3StatsTable etherlike-mib/dot3StatsTable/dot3StatsTable_data_access etherlike-mib/dot3StatsTable/dot3StatsTable_data_get etherlike-mib/dot3StatsTable/dot3StatsTable_data_set etherlike-mib/dot3StatsTable/dot3StatsTable_interface hardware/cpu hardware/cpu/cpu hardware/cpu/cpu_linux hardware/fsys hardware/fsys/fsys_mntent hardware/fsys/hw_fsys hardware/memory hardware/memory/hw_mem hardware/memory/memory_linux hardware/sensors hardware/sensors/hw_sensors hardware/sensors/lmsensors_v3 header_complex host host/data_access/swinst host/data_access/swrun host/data_access/swrun_procfs_status host/hrSWInstalledTable host/hrSWRunPerfTable host/hrSWRunTable host/hr_device host/hr_disk host/hr_network host/hr_other host/hr_partition host/hr_print host/hr_proc host/hr_system host/hrh_filesys host/hrh_storage if-mib if-mib/data_access/interface if-mib/data_access/interface_ioctl if-mib/data_access/interface_linux if-mib/ifTable if-mib/ifTable/ifTable if-mib/ifTable/ifTable_data_access if-mib/ifTable/ifTable_interface if-mib/ifXTable if-mib/ifXTable/ifXTable if-mib/ifXTable/ifXTable_data_access if-mib/ifXTable/ifXTable_interface ip-forward-mib ip-forward-mib/data_access/route ip-forward-mib/data_access/route_common ip-forward-mib/data_access/route_ioctl ip-forward-mib/data_access/route_linux ip-forward-mib/inetCidrRouteTable ip-forward-mib/inetCidrRouteTable/inetCidrRouteTable ip-forward-mib/inetCidrRouteTable/inetCidrRouteTable_data_access ip-forward-mib/inetCidrRouteTable/inetCidrRouteTable_interface ip-forward-mib/ipCidrRouteTable ip-forward-mib/ipCidrRouteTable/ipCidrRouteTable ip-forward-mib/ipCidrRouteTable/ipCidrRouteTable_data_access ip-forward-mib/ipCidrRouteTable/ipCidrRouteTable_interface ip-mib ip-mib/data_access/arp ip-mib/data_access/arp_common ip-mib/data_access/arp_netlink ip-mib/data_access/defaultrouter ip-mib/data_access/defaultrouter_common ip-mib/data_access/defaultrouter_linux ip-mib/data_access/ipaddress ip-mib/data_access/ipaddress_common ip-mib/data_access/ipaddress_ioctl ip-mib/data_access/ipaddress_linux ip-mib/data_access/ipv6scopezone ip-mib/data_access/ipv6scopezone_common ip-mib/data_access/ipv6scopezone_linux ip-mib/data_access/scalars_common ip-mib/data_access/scalars_linux ip-mib/data_access/systemstats ip-mib/data_access/systemstats_common ip-mib/data_access/systemstats_linux ip-mib/inetNetToMediaTable ip-mib/inetNetToMediaTable/inetNetToMediaTable ip-mib/inetNetToMediaTable/inetNetToMediaTable_data_access ip-mib/inetNetToMediaTable/inetNetToMediaTable_interface ip-mib/ipAddressPrefixTable ip-mib/ipAddressPrefixTable/ipAddressPrefixTable ip-mib/ipAddressPrefixTable/ipAddressPrefixTable_data_access ip-mib/ipAddressPrefixTable/ipAddressPrefixTable_interface ip-mib/ipAddressTable ip-mib/ipAddressTable/ipAddressTable ip-mib/ipAddressTable/ipAddressTable_data_access ip-mib/ipAddressTable/ipAddressTable_interface ip-mib/ipDefaultRouterTable ip-mib/ipDefaultRouterTable/ipDefaultRouterTable ip-mib/ipDefaultRouterTable/ipDefaultRouterTable_data_access ip-mib/ipDefaultRouterTable/ipDefaultRouterTable_data_get ip-mib/ipDefaultRouterTable/ipDefaultRouterTable_interface ip-mib/ipIfStatsTable ip-mib/ipIfStatsTable/ipIfStatsTable ip-mib/ipIfStatsTable/ipIfStatsTable_data_access ip-mib/ipIfStatsTable/ipIfStatsTable_data_get ip-mib/ipIfStatsTable/ipIfStatsTable_interface ip-mib/ipSystemStatsTable ip-mib/ipSystemStatsTable/ipSystemStatsTable ip-mib/ipSystemStatsTable/ipSystemStatsTable_data_access ip-mib/ipSystemStatsTable/ipSystemStatsTable_interface ip-mib/ip_scalars ip-mib/ipv4InterfaceTable ip-mib/ipv4InterfaceTable/ipv4InterfaceTable ip-mib/ipv4InterfaceTable/ipv4InterfaceTable_data_access ip-mib/ipv4InterfaceTable/ipv4InterfaceTable_interface ip-mib/ipv6InterfaceTable ip-mib/ipv6InterfaceTable/ipv6InterfaceTable ip-mib/ipv6InterfaceTable/ipv6InterfaceTable_data_access ip-mib/ipv6InterfaceTable/ipv6InterfaceTable_interface ip-mib/ipv6ScopeZoneIndexTable ip-mib/ipv6ScopeZoneIndexTable/ipv6ScopeZoneIndexTable ip-mib/ipv6ScopeZoneIndexTable/ipv6ScopeZoneIndexTable_data_access ip-mib/ipv6ScopeZoneIndexTable/ipv6ScopeZoneIndexTable_interface mibII mibII/at mibII/data_access/at_linux mibII/data_access/at_unix mibII/icmp mibII/ifTable mibII/ip mibII/ipAddr mibII/ipv6 mibII/kernel_linux mibII/mta_sendmail mibII/route_write mibII/setSerialNo mibII/snmp_mib mibII/sysORTable mibII/system_mib mibII/tcp mibII/tcpTable mibII/udp mibII/udpTable mibII/updates mibII/vacm_conf mibII/vacm_context mibII/vacm_vars mibII/var_route misc/ipfwacc notification notification-log-mib notification-log-mib/notification_log notification/snmpNotifyFilterProfileTable notification/snmpNotifyFilterProfileTable_data notification/snmpNotifyTable notification/snmpNotifyTable_data rmon-mib rmon-mib/data_access/etherstats rmon-mib/data_access/etherstats_linux rmon-mib/etherStatsTable rmon-mib/etherStatsTable/etherStatsTable rmon-mib/etherStatsTable/etherStatsTable_data_access rmon-mib/etherStatsTable/etherStatsTable_data_get rmon-mib/etherStatsTable/etherStatsTable_data_set rmon-mib/etherStatsTable/etherStatsTable_interface smux smux/smux snmp-notification-mib snmp-notification-mib/snmpNotifyFilterTable snmp-notification-mib/snmpNotifyFilterTable/snmpNotifyFilterTable snmp-notification-mib/snmpNotifyFilterTable/snmpNotifyFilterTable_data_access snmp-notification-mib/snmpNotifyFilterTable/snmpNotifyFilterTable_data_storage snmp-notification-mib/snmpNotifyFilterTable/snmpNotifyFilterTable_interface snmp-usm-dh-objects-mib snmp-usm-dh-objects-mib/usmDHParameters snmp-usm-dh-objects-mib/usmDHParameters/usmDHParameters snmp-usm-dh-objects-mib/usmDHUserKeyTable snmp-usm-dh-objects-mib/usmDHUserKeyTable/usmDHUserKeyTable snmp-usm-dh-objects-mib/usmDHUserKeyTable/usmDHUserKeyTable_data_access snmp-usm-dh-objects-mib/usmDHUserKeyTable/usmDHUserKeyTable_data_get snmp-usm-dh-objects-mib/usmDHUserKeyTable/usmDHUserKeyTable_data_set snmp-usm-dh-objects-mib/usmDHUserKeyTable/usmDHUserKeyTable_interface snmpv3/snmpEngine snmpv3/snmpMPDStats snmpv3/usmConf snmpv3/usmStats snmpv3/usmUser snmpv3mibs target target/snmpTargetAddrEntry target/snmpTargetAddrEntry_data target/snmpTargetParamsEntry target/snmpTargetParamsEntry_data target/target target/target_counters tcp-mib tcp-mib/data_access/tcpConn tcp-mib/data_access/tcpConn_common tcp-mib/data_access/tcpConn_linux tcp-mib/tcpConnectionTable tcp-mib/tcpConnectionTable/tcpConnectionTable tcp-mib/tcpConnectionTable/tcpConnectionTable_data_access tcp-mib/tcpConnectionTable/tcpConnectionTable_interface tcp-mib/tcpListenerTable tcp-mib/tcpListenerTable/tcpListenerTable tcp-mib/tcpListenerTable/tcpListenerTable_data_access tcp-mib/tcpListenerTable/tcpListenerTable_interface testhandler tlstm-mib tlstm-mib/snmpTlstmAddrTable tlstm-mib/snmpTlstmAddrTable/snmpTlstmAddrTable tlstm-mib/snmpTlstmCertToTSNTable tlstm-mib/snmpTlstmCertToTSNTable/snmpTlstmCertToTSNTable tlstm-mib/snmpTlstmParamsTable tlstm-mib/snmpTlstmParamsTable/snmpTlstmParamsTable tlstm-mib/snmpTlstmSession tlstm-mib/snmpTlstmSession/snmpTlstmSession tsm-mib tsm-mib/snmpTsmConfigurationUsePrefix tsm-mib/snmpTsmStats tsm-mib/snmpTsmStats/snmpTsmStats tunnel tunnel/tunnel ucd-snmp/disk_hw ucd-snmp/diskio ucd-snmp/dlmod ucd-snmp/errormib ucd-snmp/extensible ucd-snmp/file ucd-snmp/lmsensorsMib ucd-snmp/loadave ucd-snmp/logmatch ucd-snmp/memory ucd-snmp/pass ucd-snmp/pass_common ucd-snmp/pass_persist ucd-snmp/proc ucd-snmp/proxy ucd-snmp/versioninfo ucd-snmp/vmstat ucd_snmp udp-mib udp-mib/data_access/udp_endpoint udp-mib/data_access/udp_endpoint_common udp-mib/data_access/udp_endpoint_linux udp-mib/udpEndpointTable udp-mib/udpEndpointTable/udpEndpointTable udp-mib/udpEndpointTable/udpEndpointTable_data_access udp-mib/udpEndpointTable/udpEndpointTable_interface util_funcs util_funcs/get_pid_from_inode util_funcs/header_generic util_funcs/header_simple_table util_funcs/restart utilities utilities/execute utilities/iquery utilities/override" \
                --with-out-mib-modules="ucd-snmp/lmSensors" \
                --with-mysql \
                --with-perl-modules \
                --without-python-modules \
                --with-security-modules="usm ksm tsm" \
                --with-systemd \
                --with-transports="UDP UDPIPv6 TCP TCPIPv6 DTLSUDP TLSTCP"
make
```

leads to:
```
/usr/bin/ld: mibgroup/disman/ping/.libs/pingProbeHistoryTable.o:git-snmp/agent/mibgroup/disman/ping/pingCtlTable.h:331: multiple definition of `status_snapshot'; mibgroup/disman/ping/.libs/pingCtlTable.o:git-snmp/agent/mibgroup/disman/ping/pingCtlTable.h:331: first defined here
/usr/bin/ld: mibgroup/disman/ping/.libs/pingProbeHistoryTable.o:git-snmp/agent/mibgroup/disman/ping/pingCtlTable.h:330: multiple definition of `exiting'; mibgroup/disman/ping/.libs/pingCtlTable.o:git-snmp/agent/mibgroup/disman/ping/pingCtlTable.h:330: first defined here
/usr/bin/ld: mibgroup/disman/ping/.libs/pingResultsTable.o:git-snmp/agent/mibgroup/disman/ping/pingCtlTable.h:331: multiple definition of `status_snapshot'; mibgroup/disman/ping/.libs/pingCtlTable.o:git-snmp/agent/mibgroup/disman/ping/pingCtlTable.h:331: first defined here
/usr/bin/ld: mibgroup/disman/ping/.libs/pingResultsTable.o:git-snmp/agent/mibgroup/disman/ping/pingCtlTable.h:330: multiple definition of `exiting'; mibgroup/disman/ping/.libs/pingCtlTable.o:git-snmp/agent/mibgroup/disman/ping/pingCtlTable.h:330: first defined here
/usr/bin/ld: .libs/mib_modules.o:git-snmp/agent/./mibgroup/disman/ping/pingCtlTable.h:331: multiple definition of `status_snapshot'; mibgroup/disman/ping/.libs/pingCtlTable.o:git-snmp/agent/mibgroup/disman/ping/pingCtlTable.h:331: first defined here
/usr/bin/ld: .libs/mib_modules.o:git-snmp/agent/./mibgroup/disman/ping/pingCtlTable.h:330: multiple definition of `exiting'; mibgroup/disman/ping/.libs/pingCtlTable.o:git-snmp/agent/mibgroup/disman/ping/pingCtlTable.h:330: first defined here
collect2: error: ld returned 1 exit status
```
Full build log:
[snmp_v5.8.1.pre2_amd64.build.log](https://github.com/net-snmp/net-snmp/files/5069554/snmp_v5.8.1.pre2_amd64.build.log)

**Comment by jer-gentoo** [comment]
This was already fixed in 46f755fe4e6c2a919eebae0c55495f5f1207d06c and released in https://github.com/net-snmp/net-snmp/releases/tag/v5.8.1.rc1 and up.

**Comment by jean-christophe-manciot** [comment]
I am able to build v5.8.1.rc1.
